src/negotiation/order.ts

```diff
@@ -1,9 +1,9 @@
 import { BigNumber } from "bignumber.js";
 import { Asset, Ledger, SwapProperties } from "../cnd";
+import { Swap } from "../swap";
 import { getToken, Token } from "../tokens/tokens";
 
-export interface Order {
-  tradingPair: string;
+export interface OrderParams {
```

Comment:
🙃 I find the design of the `Order` a bit confusing. From code

```
export class Order {
  constructor(
    public readonly orderParams: OrderParams,
    public readonly criteria: TakerCriteria,
    public readonly takeOrder: (
      orderParams: OrderParams
    ) => Promise<Swap | undefined>
  ) {}
```

So the order consists of `OrderParams` and a `TakerCriteria`, why is the abstraction of `OrderParams` needed? What exactly is a `TakerCriteria`?

So it is cleaner to separate it from the `Order`? I feel there is quite a lot of classes now and it's not immediately understandable why and what they are for. 
Motivation for writing this: Started to write comments for code documentation and it's not that easy to explain what is going on. What is currently there works, but I wonder if we could design it so it is easier to understand. 

Can we simplify this by integrating the `OrderParams` into the `Order`? Is this abstraction really needed?

Reply:
Please note that the lib consumer is not supposed to constructor `Order` by themselves.
They would only use `OrderParams` (a simple object) to add orders to the order book.

Interesting point about documentation because this means that the constructor of `Order` should not be documented for example.

Reply:
I think I understand what you are doing, but I still find the naming a bit confusing. I don't have a perfect solution in mind yet, but maybe we should step back, take a whiteboard and design this outside the code. The way `Order` and `OrderParams` is introduced in this PR is a bit confusing for me. we have a lot of variables named `order` now, that are actually `OrderParams`. Maybe a simple rename can fix this, not sure.